Reference/using-ioc.md

```diff
@@ -141,3 +141,82 @@ If you don't register assemblies that contain controllers you may end up with YS
 ##Things to note
 
 We use a custom MVC controller builder in our code called `Umbraco.Web.Mvc.MasterControllerFactory`, which needs to always be the default controller factory, if you change this Umbraco will probably not work anymore. The good news is that you can specify 'slave' factories so you can specify custom controller factories for different purposes. You would just need to create a new class that inherits from `Umbraco.Web.Mvc.IFilteredControllerFactory` and ensure that the class is public (so it can be found). If your IoC implementation affects the default controller factory, you may have to modify it in order to support this implementation. For the most part, most IoC frameworks will just target setting a custom DependencyResolver which is 100% ok.
+
+##Unity Example
+
+Install the Unity, Unity.Mvc and Unity.AspNet.WebApi packages.
+Remove the Mvc and Api specific files created, we will be merging those files shortly.
+Leave the UnityConfig file as is.
+
+Now create the following files to configure Unity correctly for an Umbraco site.
+
+	class UnityEvents : IApplicationEventHandler
+	{
+	    public void OnApplicationStarted(
+		UmbracoApplicationBase httpApplication,
+		ApplicationContext applicationContext
+	    )
+	    {
+		var container = UnityConfig.GetConfiguredContainer();
+
+		// Web API
+		GlobalConfiguration.Configuration.DependencyResolver 
+			= new Microsoft.Practices.Unity.WebApi.UnityDependencyResolver(container);
+		// MVC
+		DependencyResolver.SetResolver(new Microsoft.Practices.Unity.Mvc.UnityDependencyResolver(container));
+
+		// This will automatically scan an assembly for classes fitting this convention:
+		// Interface name: IMyClass
+		// Class name: MyClass
+		// and automatically register those types for you.
+		// If you do decide to use this feature, make sure it is the first one called.
+		// Conflicting registrations that follow will override previous ones.
+		container.RegisterTypes(
+		    AllClasses.FromAssemblies(typeof(UnityEvents).Assembly),
+		    WithMappings.FromMatchingInterface,
+		    WithName.Default
+		);
+
+		// The UmbracoContext must be registered so that the umbraco backoffice controllers 
+		// can be successfully resolved
+		container.RegisterType<UmbracoContext>(
+			new PerRequestLifetimeManager(), 
+			new InjectionFactory(c => UmbracoContext.Current)
+		);
+
+		// Instantiate the HealthCheckController using it's default parameterless constructor
+		container.RegisterType<HealthCheckController>(new InjectionConstructor());
```

Comment:
Why are these controllers registered singularly? These docs should register all MVC and WebApi controllers found in the `typeof(UmbracoApplication).Assembly` just like the autofac docs.  The code block above does this but I'm unsure if that's for MVC or for WebAPI (or both) ? Similarly the LegacyTreeController doesn't need to be registered singularly, it will be taken care of by registering all MVC and WebApi controllers found in `typeof(UmbracoApplication).Assembly`